src/lsm/lsm_cursor.c

```diff
@@ -902,6 +944,93 @@ err:	__clsm_leave(clsm);
 }
 
 /*
+ * __clsm_random_chunk --
+ *	Pick a chunk at random, weighted by the size of all chunks. Weighting
+ * proportional to documents avoids biasing towards small chunks. Then return
+ * the cursor on the chunk we have picked.
+ */
+static int
+__clsm_random_chunk(WT_SESSION_IMPL *session,
+    WT_CURSOR_LSM *clsm, WT_CURSOR **cursor)
+{
+	uint64_t checked_docs, i, rand_doc, total_docs;
+
+	rand_doc = __wt_random(&session->rnd);
+	/*
+	 * If the tree is empty we cannot do a random lookup, so return a
+	 * WT_NOTFOUND.
+	 */
+	if (clsm->nchunks == 0)
+		return (WT_NOTFOUND);
+	for (total_docs = i = 0; i < clsm->nchunks; i++) {
+		total_docs += clsm->iters[i]->count;
+	}
+	if (total_docs == 0)
+		return (WT_NOTFOUND);
+
```

Comment:
Now that you are not taking locks, maybe you can move random number generation after checking for "total_docs == 0". Can also shorten it to:
rand_doc = __wt_random(&session->and) % total_docs;